src/commonMain/kotlin/exchange.dydx.abacus/state/manager/StateManagerAdaptor.kt

```diff
@@ -2398,29 +2416,98 @@ open class StateManagerAdaptor(
+                val signedMessage = parser.asString(result["signedMessage"])
+                val publicKey = parser.asString(result["publicKey"])
+                val timestamp = parser.asString(result["timestamp"])
+
+                val isUrlAndKeysPresent =
+                    url != null && signedMessage != null && publicKey != null && timestamp != null
+                val isStatusValid = status != ComplianceStatus.UNKNOWN
+
+                if (isUrlAndKeysPresent && isStatusValid) {
+                    val body: IMap<String, String> = iMapOf(
+                        "address" to address,
+                        "message" to message,
+                        "currentStatus" to status.toString(),
+                        "action" to action.toString(),
+                        "signedMessage" to signedMessage!!,
+                        "pubkey" to publicKey!!,
+                        "timestamp" to timestamp!!,
+                    )
+                    val header = iMapOf(
+                        "Content-Type" to "application/json",
+                    )
+                    post(
+                        url!!,
+                        header,
+                        body.toJsonPrettyPrint(),
+                        callback = { _, response, httpCode, _ ->
+                            handleComplianceResponse(response, httpCode)
+                        },
+                    )
+                } else {
+                    compliance = Compliance(compliance?.geo, ComplianceStatus.UNKNOWN)
+                }
+            } else {
+                compliance = Compliance(compliance?.geo, ComplianceStatus.UNKNOWN)
+            }
+        }
     }
 
-    open fun complianceScreen(address: String) {
-        val url = complianceScreenUrl(address)
+    private fun complianceScreen(address: Address) {
+        val url = complianceScreenUrl(address.rawAddress)
         if (url != null) {
             get(
                 url,
                 null,
                 null,
                 callback = { _, response, httpCode, _ ->
-                    compliance = if (success(httpCode) && response != null) {
-                        val payload = parser.decodeJsonObject(response)?.toIMap()
-                        if (payload != null) {
-                            val status = parser.asString(payload["status"])
-                            val complianceStatus = ComplianceStatus.invoke(status) ?: ComplianceStatus.UNKNOWN
-                            Compliance(compliance?.geo, complianceStatus)
-                        } else {
-                            Compliance(compliance?.geo, ComplianceStatus.UNKNOWN)
-                        }
-                    } else {
-                        Compliance(compliance?.geo, ComplianceStatus.UNKNOWN)
-                    }
+                    val complianceStatus = handleComplianceResponse(response, httpCode)
+                    updateCompliance(address.rawAddress, complianceStatus)
```

Comment:
i think we forgot to check `DydxAddress` vs `EvmAddress`? To enforce this, we can change the signature of `updateCompliance()` to take in `DydxAddress` instead of `String`.